### PR TITLE
Add a smoke test for published repository

### LINKED ISSUE #38: feature: watchdog action
There could be an independent failure watchdog action in addition to #37:

* on a cron workflow , run an action that uses a TUF client to fetch the metadata from the published location
* file an issue if refresh fails
* file an issue if online roles are within signing-period: in other words there should have been a version bump already (note that this requires some nuance as online signing happens in cron by default every 6 hours -- so currently we may be 6 hours into the signing period before the signing happens)
* (maybe) file an issue if any delegated offline role has expired
* (maybe) file an issue if offline role is in signing period and a signing event does not exist

### PR DESCRIPTION
**Pull request by jku**
This adds an action that runs a basic client against the repository  

The action could be used in
* cron workflow
* publish workflow

Fixes #38 (at least parts of it)

### CODE REVIEW

**Comment by jku** [comment]
Test run
https://github.com/jku/tuf-on-ci-sigstore-test/actions/runs/7585094401

**Comment by jku** [comment]
> I have no idea if I should expect that to fail very often when we publish to GH Pages and immediately run this...

Based on running a test over the weekend:
* 200 successful publishes and immediate fetches from Pages
* 2 failures, both GitHub outage related
* failures could not have been avoided by retries

 I think I'd be fine adding this in without any retry support: it seems to typically work and #168 is the generics olution to ephemeral failures  -- if issues come up we can add retries or something afterwards.

Mapping out how this can be used in the workflows:
1. We'll want a new workflow anyway: a cron triggered smoke test workflow that calls this action
1. It would be great if the same workflow could be automatically triggered by publishing to pages. I don't think that is possible without another workflow dispatch...
1. So likely publish workflow should just call the same action -- it makes publish more complicated but still readable

**Comment by jku** [comment]
Marked this ready for review: It's a self contained new feature that does not affect users until some workflows start to use the action.

File: actions/test-repository/action.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
+name: 'Test TUF-on-CI repository'
+description: 'Test a published TUF-on-CI repository with a client'
+
+inputs:
+  metadata_url:
+    description: 'base metadata URL the client should use'
+    default: ''
+  artifact_url:
+    description: 'Base artifact URL the client should use'
+    default: ''
```
Comment (kommendorkapten) [inline actions/test-repository/action.yml:10]: How about an optional third parameter that would be a list of targets to download as well? I would think about making it simple, i.e only test for existence, not for verifying content (i.e the message digest).

**Comment by kommendorkapten** [comment]
Another option for robustness is to extract the latest version of the top level metadata and provide that to verify that the correct version is deployed (can be done in a follow up PR).

**Review by kommendorkapten — APPROVED** [review]

File: actions/test-repository/action.yml (hunk 1 above)
Comment (jku) [reply actions/test-repository/action.yml:10]: Yeah I agree it makes sense: left it out of this version to keep it short but I could do your simplified suggestion...

**Comment by jku** [comment]
> Another option for robustness is to extract the latest version of the top level metadata and provide that to verify that the correct version is deployed (can be done in a follow up PR).

I think the freshness check that is currently in there is nice because you don't have to provide arguments: the argument is essentially the repository ref that we are on and tuf-on-ci-test-client can figure out the correct versions from files on disk.
Currently the freshness test in the end only checks that root and timestamp match the current on-disk versions: this could be expanded if there is a use case this does not cover.

Or did I misunderstand what you meant

**Comment by kommendorkapten** [comment]
No, you are right, I forgot that we updated publish action with an option to take on the git ref. So disregard that comment.

File: actions/test-repository/action.yml (hunk 1 above)
Comment (jku) [reply actions/test-repository/action.yml:10]: I've added an optional `expected_artifact` argument. I don't know if it's actually useful for most repos but it'll likely be useful for root-signing and it's not much more complicated.

**Comment by jku** [comment]
I think I'll wait until 0.4 is released before merging: We don't have the workflows ready yet for this so no need to have the action in tuf-on-ci yet either

**Comment by jku** [comment]
merged after 0.4  release

**Pull request merged**